sdk/ml/azure-ai-ml/azure/ai/ml/_schema/_endpoint/endpoint.py

```diff
@@ -4,20 +4,24 @@
 
 import logging
 
-from marshmallow import fields
+from marshmallow import fields, validate
 
 from azure.ai.ml._restclient.v2022_02_01_preview.models import EndpointAuthMode
 from azure.ai.ml._schema.core.fields import NestedField, StringTransformedEnum
 from azure.ai.ml._schema.core.schema import PathAwareSchema
 from azure.ai.ml._schema.identity import IdentitySchema
 from azure.ai.ml._utils.utils import camel_to_snake
+from azure.ai.ml.constants._endpoint import OnlineEndpointConfigurations
 
 module_logger = logging.getLogger(__name__)
 
 
 class EndpointSchema(PathAwareSchema):
     id = fields.Str()
-    name = fields.Str(required=True)
+    name = fields.Str(
+            required=True,
+            validate=validate.Regexp(OnlineEndpointConfigurations.NAME_REGEX_PATTERN)
```

Comment:
nit: If this regex is applying for all endpoints (batch, online), perhaps makes sense to use a different class just named EndpointConfiguration.NAME_REGEX_PATTERN